src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplate.cs

```diff
@@ -84,54 +87,54 @@ public static BindingTemplate FromString(string input, bool ignoreCase = false)
         /// <returns>Resolved string if succeeded.</returns>
         /// <exception cref="InvalidOperationException">Thrown when required parameter value is not available.
         /// </exception>
-        public string Bind(IReadOnlyDictionary<string, string> parameters)
+        public string Bind(IReadOnlyDictionary<string, object> parameters)
         {
             StringBuilder builder = new StringBuilder();
 
             if (_ignoreCase && parameters != null)
             {
                 // convert to a case insensitive dictionary
-                var caseInsensitive = new Dictionary<string, string>(parameters.Count, StringComparer.OrdinalIgnoreCase);
+                var caseInsensitive = new Dictionary<string, object>(parameters.Count, StringComparer.OrdinalIgnoreCase);
```

Comment:
Good idea about improving the copying, but changing the trigger contracts is a larger change out of scope for this. Happy to tackle that separately. 


---
In reply to: [112492894](https://github.com/Azure/azure-webjobs-sdk/pull/1117#discussion_r112492894) [](ancestors = 112492894)